File: java-project-euler-100/src/main/java/net/talaatharb/projecteuler100/P005.java
package net.talaatharb.projecteuler100;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

/**
 * P005
 * 
 * @author mharb
 *
 */
public class P005  extends Solution {

	/**
	 * Problem solution entry point
	 * 
	 * @param args
	 */
	public static void main(String[] args) {
		new P005().run();
	}

	@Override
	public Number solve() {
		problemNumber = 5;
		final int last = 20;
		final List<Integer> sequence = new ArrayList<>();
		int result = 1;
		int number = 2;
		int counter = number;
		while (counter <= last) {
			for (Iterator<Integer> iterator = sequence.iterator(); iterator.hasNext();) {
				int i = iterator.next();
				if (number % i == 0) {
					number /= i;
				}
			}

			result *= number;
			sequence.add(number);
			counter += 1;
			number = counter;
		}
		
		return result;
	}

}

File: go-project-euler-100/p005.go
package main

/*P005Solve solution for p005*/
func P005Solve() int {
	const last = 20
	var sequence []int = make([]int, 0, 19)
	result := 1
	number := 2
	counter := number
	for counter <= last {
		var i, j int
		for i = 0; i < len(sequence); i++ {
			j = sequence[i]
			if number%j == 0 {
				number = number / j
			}
		}

		result *= number
		sequence = append(sequence, number)
		counter++
		number = counter
	}
	return result
}
